### DISCUSSION TITLE
How to search or apply text transformation of math text?

### DISCUSSION THREAD

**Discussion by TimeTravelPenguin** [typst/typst#4951] [Q&A]
I am going to start with [this](https://github.com/typst/typst/issues/2093#issuecomment-1784080459) example of a non-math case:
```typ
#let re = regex("(\d+).(\d+).")
#show re: it => {
  let (d, m) = it.text.match(re).captures
  [#d;th day of the #m;th month]
}

The 29.10. is a date. 
// -> The 29th day of the 10th month is a date.
```

This works as expected for standard text, but not for mathematics. Math environments do not allow you to use `show regex(...)`. Furthermore, `show math.equation: it => ...` results in content with a field of children, where each character is its own content. E.g., `$x + y = z$` results in:
```typ
equation(
  block: false,
  numbering: none,
  number-align: end + horizon,
  supplement: [Equation],
  body: [([x], [ ], [+], [ ], [y], [ ], [=], [ ], [z])],
)
```

Originally, I thought to do something like this (note that my case involves replacement):
```typ
#let re = regex("...")
#show math.equation: it => {
  let matches = it.body.children.map(child => child.text).join().matches(re)
  if matches.len() == 0 {
    return it
  }

  it.body.children.map(child => child.text).join().replace(
    re,
    m => /* do the replacement here */,
  )
}
```

However, this kind of approach is difficult to work with (ignoring the fact that the returned type is incorrect). I don't want to continue with this kind of implementation because I am unable to find any documentation regarding the semantics of what does and doesn't become single-character content.

To further extend on this issue, I may want to query a substring within my math environment (using regex). As of now, this would require my own implementation of a string-search algorithm to access the text in the content. Plus, if the content *isn't* text or contains more than one text, the algorithm becomes significantly more complex.

Is there a better approach to text query/replacement within math environments?

Edit: Note that sometimes my code doesn't even compile lol

**Comment by SylvanFranklin** [comment]
I'm also interested in this
